### ISSUE TITLE
Impossible to delete/dismiss some calendar events in Thunderbird

### ISSUE THREAD

**Issue by Ri0n** [mguessan/davmail#22]
Probably it's related to recently closed bug with invalid events without end date.
From OWA web interface it looks pretty much valid having start and duration/end.

```
2019-01-31 14:20:35,812 ERROR [CaldavConnection-47806] davmail  - ErrorInvalidPropertyDelete The delete action is not supported for this property. FieldURI: calendar:End
 request: <soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/" xmlns:t="http://schemas.microsoft.com/exchange/services/2006/types" xmlns:m="http://schemas.microsoft.com/exchange/services/2006/messages"><soap:Header><t:RequestServerVersion Version="Exchange2013_SP1"/><t:TimeZoneContext><t:TimeZoneDefinition Id="Belarus Standard Time"/></t:TimeZoneContext></soap:Header><soap:Body><m:UpdateItem MessageDisposition="SaveOnly" ConflictResolution="AutoResolve" SendMeetingInvitationsOrCancellations="SendToNone"><m:ItemChanges><t:ItemChange><t:ItemId Id="AAMkAGMyZWQ5ZmI2LWRhYzgtNDg4OS04YTlhLTFhNjE2OWYyNDA4NQBGAAAAAADh3GGfRBTyT4IFu4A5zCn0BwC6b8gUZ7T6RpiC3HsXRMgRAAAAAAENAAC6b8gUZ7T6RpiC3HsXRMgRAADHFuj+AAA=" ChangeKey="DwAAABYAAAC6b8gUZ7T6RpiC3HsXRMgRAADKzAmq"/><t:Updates><t:SetItemField><t:FieldURI FieldURI="calendar:Start"/><t:CalendarItem><t:Start>2019-01-25T18:00:00</t:Start></t:CalendarItem></t:SetItemField><t:DeleteItemField><t:FieldURI FieldURI="calendar:End"/></t:DeleteItemField><t:SetItemField><t:FieldURI FieldURI="calendar:StartTimeZone"/><t:CalendarItem><t:StartTimeZone Id="Russian Standard Time"></t:StartTimeZone></t:CalendarItem></t:SetItemField><t:SetItemField><t:FieldURI FieldURI="calendar:EndTimeZone"/><t:CalendarItem><t:EndTimeZone Id="Russian Standard Time"></t:EndTimeZone></t:CalendarItem></t:SetItemField><t:SetItemField><t:FieldURI FieldURI="calendar:IsAllDayEvent"/><t:CalendarItem><t:IsAllDayEvent>false</t:IsAllDayEvent></t:CalendarItem></t:SetItemField><t:SetItemField><t:FieldURI FieldURI="item:Sensitivity"/><t:Item><t:Sensitivity>Normal</t:Sensitivity></t:Item></t:SetItemField><t:DeleteItemField><t:ExtendedFieldURI PropertyTag="0x1000" PropertyType="String"/></t:DeleteItemField><t:DeleteItemField><t:ExtendedFieldURI PropertyTag="0x37" PropertyType="String"/></t:DeleteItemField><t:DeleteItemField><t:FieldURI FieldURI="calendar:Location"/></t:DeleteItemField><t:DeleteItemField><t:FieldURI FieldURI="calendar:RequiredAttendees"/></t:DeleteItemField><t:DeleteItemField><t:FieldURI FieldURI="calendar:OptionalAttendees"/></t:DeleteItemField><t:SetItemField><t:FieldURI FieldURI="item:ReminderIsSet"/><t:Item><t:ReminderIsSet>true</t:ReminderIsSet></t:Item></t:SetItemField><t:SetItemField><t:FieldURI FieldURI="item:ReminderMinutesBeforeStart"/><t:Item><t:ReminderMinutesBeforeStart>2880</t:ReminderMinutesBeforeStart></t:Item></t:SetItemField><t:SetItemField><t:ExtendedFieldURI DistinguishedPropertySetId="PublicStrings" PropertyName="xmozlastack" PropertyType="String"/><t:Item><t:ExtendedProperty><t:ExtendedFieldURI DistinguishedPropertySetId="PublicStrings" PropertyName="xmozlastack" PropertyType="String"/><t:Value>20190131T112032Z</t:Value></t:ExtendedProperty></t:Item></t:SetItemField></t:Updates></t:ItemChange></m:ItemChanges></m:UpdateItem></soap:Body></soap:Envelope>
davmail.exchange.ews.EWSException: ErrorInvalidPropertyDelete The delete action is not supported for this property. FieldURI: calendar:End
 request: <soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/" xmlns:t="http://schemas.microsoft.com/exchange/services/2006/types" xmlns:m="http://schemas.microsoft.com/exchange/services/2006/messages"><soap:Header><t:RequestServerVersion Version="Exchange2013_SP1"/><t:TimeZoneContext><t:TimeZoneDefinition Id="Belarus Standard Time"/></t:TimeZoneContext></soap:Header><soap:Body><m:UpdateItem MessageDisposition="SaveOnly" ConflictResolution="AutoResolve" SendMeetingInvitationsOrCancellations="SendToNone"><m:ItemChanges><t:ItemChange><t:ItemId Id="AAMkAGMyZWQ5ZmI2LWRhYzgtNDg4OS04YTlhLTFhNjE2OWYyNDA4NQBGAAAAAADh3GGfRBTyT4IFu4A5zCn0BwC6b8gUZ7T6RpiC3HsXRMgRAAAAAAENAAC6b8gUZ7T6RpiC3HsXRMgRAADHFuj+AAA=" ChangeKey="DwAAABYAAAC6b8gUZ7T6RpiC3HsXRMgRAADKzAmq"/><t:Updates><t:SetItemField><t:FieldURI FieldURI="calendar:Start"/><t:CalendarItem><t:Start>2019-01-25T18:00:00</t:Start></t:CalendarItem></t:SetItemField><t:DeleteItemField><t:FieldURI FieldURI="calendar:End"/></t:DeleteItemField><t:SetItemField><t:FieldURI FieldURI="calendar:StartTimeZone"/><t:CalendarItem><t:StartTimeZone Id="Russian Standard Time"></t:StartTimeZone></t:CalendarItem></t:SetItemField><t:SetItemField><t:FieldURI FieldURI="calendar:EndTimeZone"/><t:CalendarItem><t:EndTimeZone Id="Russian Standard Time"></t:EndTimeZone></t:CalendarItem></t:SetItemField><t:SetItemField><t:FieldURI FieldURI="calendar:IsAllDayEvent"/><t:CalendarItem><t:IsAllDayEvent>false</t:IsAllDayEvent></t:CalendarItem></t:SetItemField><t:SetItemField><t:FieldURI FieldURI="item:Sensitivity"/><t:Item><t:Sensitivity>Normal</t:Sensitivity></t:Item></t:SetItemField><t:DeleteItemField><t:ExtendedFieldURI PropertyTag="0x1000" PropertyType="String"/></t:DeleteItemField><t:DeleteItemField><t:ExtendedFieldURI PropertyTag="0x37" PropertyType="String"/></t:DeleteItemField><t:DeleteItemField><t:FieldURI FieldURI="calendar:Location"/></t:DeleteItemField><t:DeleteItemField><t:FieldURI FieldURI="calendar:RequiredAttendees"/></t:DeleteItemField><t:DeleteItemField><t:FieldURI FieldURI="calendar:OptionalAttendees"/></t:DeleteItemField><t:SetItemField><t:FieldURI FieldURI="item:ReminderIsSet"/><t:Item><t:ReminderIsSet>true</t:ReminderIsSet></t:Item></t:SetItemField><t:SetItemField><t:FieldURI FieldURI="item:ReminderMinutesBeforeStart"/><t:Item><t:ReminderMinutesBeforeStart>2880</t:ReminderMinutesBeforeStart></t:Item></t:SetItemField><t:SetItemField><t:ExtendedFieldURI DistinguishedPropertySetId="PublicStrings" PropertyName="xmozlastack" PropertyType="String"/><t:Item><t:ExtendedProperty><t:ExtendedFieldURI DistinguishedPropertySetId="PublicStrings" PropertyName="xmozlastack" PropertyType="String"/><t:Value>20190131T112032Z</t:Value></t:ExtendedProperty></t:Item></t:SetItemField></t:Updates></t:ItemChange></m:ItemChanges></m:UpdateItem></soap:Body></soap:Envelope>
        at davmail.exchange.ews.EWSMethod.checkSuccess(EWSMethod.java:778)
        at davmail.exchange.ews.EwsExchangeSession.internalExecuteMethod(EwsExchangeSession.java:3007)
        at davmail.exchange.ews.EwsExchangeSession.executeMethod(EwsExchangeSession.java:2971)
        at davmail.exchange.ews.EwsExchangeSession$Event.createOrUpdate(EwsExchangeSession.java:2135)
        at davmail.exchange.ews.EwsExchangeSession.internalCreateOrUpdateEvent(EwsExchangeSession.java:2710)
        at davmail.exchange.ExchangeSession.createOrUpdateItem(ExchangeSession.java:3211)
        at davmail.caldav.CaldavConnection.handleFolderOrItem(CaldavConnection.java:290)
        at davmail.caldav.CaldavConnection.handleRequest(CaldavConnection.java:228)
        at davmail.caldav.CaldavConnection.run(CaldavConnection.java:171)
```

**Comment by mguessan** [comment]
Ok, we need the full Caldav message (with VCALENDAR body) to investigate

**Comment by Ri0n** [comment]
Can you help me with that in Thunderbird? Maybe there is some debugging extensions/logs?
I don't see anything like this in context menu of the event.

**Comment by mguessan** [comment]
You have everything in davmail.log, for more info on Thunderbird, check console log, see http://davmail.sourceforge.net/faq.html

**Comment by Ri0n** [comment]
oh. I just synced the calendar and this broken event has disappeared..
maybe it was removed by someone else. anyway, I can't provide debug info now :(

**Comment by Ri0n** [comment]
strange it just appeared again. and I still can't delete/dismiss it.

Here is what I got from logs on "Save And Close" button from event edit dialog
```
2019-02-04 18:08:42,181 DEBUG [CaldavConnection-57100] davmail.caldav.CaldavConnection  - < BEGIN:VCALENDAR
PRODID:-//Mozilla.org/NONSGML Mozilla Calendar V1.1//EN
VERSION:2.0
BEGIN:VTIMEZONE
TZID:Europe/Minsk
BEGIN:STANDARD
TZOFFSETFROM:+0300
TZOFFSETTO:+0300
TZNAME:+03
DTSTART:19700101T000000
END:STANDARD
END:VTIMEZONE
BEGIN:VEVENT
CREATED:20190204T150042Z
LAST-MODIFIED:20190204T150842Z
DTSTAMP:20190204T150842Z
UID:040000008200E00074C5B7101A82E00800000000B0529711DF91D20100000000000000
 0010000000C9EDD3D87160054D887FB4D7EBDF0AF4
RDATE;VALUE=DATE-TIME:20190125T150000Z
DTSTART;TZID=Europe/Minsk:20190125T180000
X-MOZ-FAKED-MASTER:1
X-MOZ-GENERATION:1
END:VEVENT
BEGIN:VEVENT
LAST-MODIFIED:20190204T150842Z
DTSTAMP:20190204T150842Z
UID:040000008200E00074C5B7101A82E00800000000B0529711DF91D20100000000000000
 0010000000C9EDD3D87160054D887FB4D7EBDF0AF4
SUMMARY:INFO-BLOX  MONTHLY HAPPY BIRTHDAY PARTY
PRIORITY:0
RECURRENCE-ID;TZID=Europe/Minsk:20190125T180000
ORGANIZER;CN=Olga;SENT-BY="mailto:sergey@epam.com":mailto:
 Olga@epam.com
ATTENDEE;CN=Olga;PARTSTAT=NEEDS-ACTION;ROLE=REQ-PARTICIPANT:mailto
 :Olga@epam.com
ATTENDEE;CN=Project;PARTSTAT=NEEDS-ACTION;ROLE=REQ-PARTICIPANT:m
 ailto:Project@epam.com
ATTENDEE;CN=Siarhei;PARTSTAT=NEEDS-ACTION;ROLE=REQ-PARTICIPANT:mailt
 o:Siarhei@epam.com
ATTENDEE;CN=Tatsiana;PARTSTAT=NEEDS-ACTION;ROLE=REQ-PARTICIPANT:m
 ailto:Tatsiana@epam.com
ATTENDEE;CN=Dzianis;PARTSTAT=NEEDS-ACTION;ROLE=REQ-PARTICIPANT:
 mailto:Dzianis@epam.com
ATTENDEE;CN=Uladzimir;PARTSTAT=NEEDS-ACTION;ROLE=REQ-PARTICIPANT:
 mailto:Uladzimir@epam.com
DTSTART;TZID=Europe/Minsk:20190201T180000
DTEND;TZID=Europe/Minsk:20190201T203000
DESCRIPTION:As a tradition we meet every last Friday of each month to cele
 brate birthdays of our colleagues.\n\nLet’s party!\n
SEQUENCE:493
LOCATION:KITCHEN BLOCK B
X-MICROSOFT-CDO-APPT-SEQUENCE:492
X-MICROSOFT-CDO-OWNERAPPTID:-941873183
X-MICROSOFT-CDO-BUSYSTATUS:BUSY
X-MICROSOFT-CDO-INTENDEDSTATUS:BUSY
X-MICROSOFT-CDO-ALLDAYEVENT:FALSE
X-MICROSOFT-CDO-IMPORTANCE:2
X-MICROSOFT-CDO-INSTTYPE:3
X-MICROSOFT-DONOTFORWARDMEETING:FALSE
X-MICROSOFT-DISALLOW-COUNTER:FALSE
X-MICROSOFT-LOCATIONS:"LocationFullAddress":""}]
BEGIN:VALARM
ACTION:DISPLAY
TRIGGER;VALUE=DURATION:-P2D
DESCRIPTION:REMINDER
END:VALARM
END:VEVENT
END:VCALENDAR
```

and this
```
2019-02-04 18:16:28,453 DEBUG [CaldavConnection-57156] davmail.exchange.ExchangeSession  - Fixed Vcalendar body to server:
BEGIN:VCALENDAR
PRODID:-//Mozilla.org/NONSGML Mozilla Calendar V1.1//EN
VERSION:2.0
METHOD:PUBLISH
BEGIN:VTIMEZONE
TZID:Russian Standard Time
BEGIN:STANDARD
TZOFFSETFROM:+0300
TZOFFSETTO:+0300
TZNAME:+03
DTSTART:19700101T000000
END:STANDARD
END:VTIMEZONE
BEGIN:VEVENT
CREATED:20190204T150042Z
LAST-MODIFIED:20190204T151628Z
DTSTAMP:20190204T151628Z
UID:040000008200E00074C5B7101A82E00800000000B0529711DF91D20100000000000000001
 0000000C9EDD3D87160054D887FB4D7EBDF0AF4
RDATE;VALUE=DATE-TIME:20190125T150000Z
DTSTART;TZID=Russian Standard Time:20190125T180000
X-MOZ-FAKED-MASTER:1
X-MOZ-GENERATION:1
ORGANIZER:MAILTO:Sergey@epam.com
X-MICROSOFT-CDO-ALLDAYEVENT:FALSE
END:VEVENT
BEGIN:VEVENT
LAST-MODIFIED:20190204T151628Z
DTSTAMP:20190204T151628Z
UID:040000008200E00074C5B7101A82E00800000000B0529711DF91D20100000000000000001
 0000000C9EDD3D87160054D887FB4D7EBDF0AF4
SUMMARY:INFO-BLOX  MONTHLY HAPPY BIRTHDAY PARTY
PRIORITY:0
RECURRENCE-ID;TZID=Russian Standard Time:20190125T180000
ORGANIZER;CN="Olga";SENT-BY="mailto:sergey@epam.com":mailto:O
 lga@epam.com
ATTENDEE;CN="Olga";PARTSTAT=NEEDS-ACTION;ROLE=REQ-PARTICIPANT:mailto:
 Olga@epam.com
ATTENDEE;CN="Project";PARTSTAT=NEEDS-ACTION;ROLE=REQ-PARTICIPANT:ma
 ilto:Project@epam.com
ATTENDEE;CN="Siarhei";PARTSTAT=NEEDS-ACTION;ROLE=REQ-PARTICIPANT:mailto
 :Siarhei@epam.com
ATTENDEE;CN="Tatsiana";PARTSTAT=NEEDS-ACTION;ROLE=REQ-PARTICIPANT:ma
 ilto:Tatsiana@epam.com
ATTENDEE;CN="Dzianis";PARTSTAT=NEEDS-ACTION;ROLE=REQ-PARTICIPANT:m
 ailto:Dzianis@epam.com
ATTENDEE;CN="Uladzimir";PARTSTAT=NEEDS-ACTION;ROLE=REQ-PARTICIPANT:m
 ailto:Uladzimir@epam.com
DTSTART;TZID=Russian Standard Time:20190201T180000
DTEND;TZID=Russian Standard Time:20190201T203000
DESCRIPTION:As a tradition we meet every last Friday of each month to celebra
 te birthdays of our colleagues.\n\nLet’s party!\n
SEQUENCE:493
LOCATION:KITCHEN BLOCK B
X-MICROSOFT-CDO-APPT-SEQUENCE:492
X-MICROSOFT-CDO-OWNERAPPTID:-941873183
X-MICROSOFT-CDO-BUSYSTATUS:BUSY
X-MICROSOFT-CDO-INTENDEDSTATUS:BUSY
X-MICROSOFT-CDO-ALLDAYEVENT:FALSE
X-MICROSOFT-CDO-IMPORTANCE:2
X-MICROSOFT-CDO-INSTTYPE:3
X-MICROSOFT-DONOTFORWARDMEETING:FALSE
X-MICROSOFT-DISALLOW-COUNTER:FALSE
X-MICROSOFT-LOCATIONS:"LocationFullAddress":""}]
X-MICROSOFT-CDO-REPLYTIME:20190204T151628Z
BEGIN:VALARM
ACTION:DISPLAY
TRIGGER;VALUE=DURATION:-P2D
DESCRIPTION:REMINDER
END:VALARM
END:VEVENT
END:VCALENDAR
```

**Comment by mguessan** [comment]
Looks like a Thunderbird issue: those events have a VEVENT without DTEND, marked X-MOZ-FAKED-MASTER.

Similar to: https://bugzilla.mozilla.org/show_bug.cgi?id=953053

**Comment by Ri0n** [comment]
I see. So I have to switch my client..
I'll try Evolution then. My colleagues told me it can work directly with MS servers with mfa.

**Comment by mguessan** [comment]
=> Implemented a workaround to update only custom Mozilla properties on dismiss

**Comment by mguessan** [comment]
Released in 5.2.0
